geo_heatmap.py

```diff
@@ -133,6 +157,40 @@ def loadZIPData(self, file_name, date_range):
             raise ValueError("unsupported extension for {!r}: only .json and .kml supported"
                 .format(file_name))
 
+    def loadAppleHealthData(self, file_name, date_range):
+        """
+        Load location data from an Apple Health Export file.
+        """
+        with zipfile.ZipFile(file_name) as zip_file:
+            namelist = zip_file.namelist()
+            all_routes = fnmatch.filter(namelist, "apple_health_export/workout-routes/*.gpx")
+            filtered_routes = [name for name in all_routes if dateInRange(posixpath.basename(name)[6:16], date_range)]
+
+            w = [Bar(), Percentage(), " ", Variable('name', width=30), " ", ETA()]
+            with ProgressBar(max_value=len(filtered_routes), widgets=w) as pb:
+                for i, name in enumerate(filtered_routes):
+                    pb.update(i, name=posixpath.basename(name))
+                    with zip_file.open(name) as read_file:
+                        self.loadGPXData(read_file, date_range, make_progress_bar=NullBar)
+
+    def loadZIPData(self, file_name, date_range):
+        """
+        Load location data from a zip file.
+
+        Supported zip files:
+         - Google Takekout
+         - Apple Health export
+        """
+        with zipfile.ZipFile(file_name) as zip_file:
+            namelist = zip_file.namelist()
+            if fnmatch.filter(namelist, "Takeout/*.html"):

```

Comment:
I don't really like that the files are detected by their names. Can't we detect them with `file` and say html is google and xml is apple or something like that?